request/request_test.go

```diff
@@ -0,0 +1,44 @@
+package request_test
+
+import (
+	"bytes"
+	"github.com/ONSdigital/go-ns/request"
+	. "github.com/smartystreets/goconvey/convey"
+	"io"
+	"net/http"
+	"testing"
+)
+
+func TestDrainBody_WithRequestBody(t *testing.T) {
+
+	Convey("Given a request with a body", t, func() {
+
+		body := bytes.NewBufferString("some body content")
+		r, err := http.NewRequest("GET", "/some/url", body)
+		So(err, ShouldBeNil)
+
+		Convey("When the DrainBody function is called", func() {
+
+			request.DrainBody(r)
+
+			Convey("The all bytes have been read from the body", func() {
+				_, err = r.Body.Read(make([]byte, 1))
+				So(err, ShouldEqual, io.EOF)
+			})
+		})
+	})
+}
+
+func TestDrainBody_WithoutRequestBody(t *testing.T) {
+
+	Convey("Given a request without a nil body", t, func() {
```

Comment:
`with a nil body`?  worth adding a test for *no body*, too, methinks

Reply:
Good point - I have added a test for this case